r/configure

```diff
@@ -65,8 +65,6 @@ PKG_TEST_HEADER="<arrow/api.h>"
 # Some env vars that control the build (all logical, case insensitive)
```

Comment:
(Commenting here because I can't add a comment higher up ^^^)

Above where it says,

```
# * Installing a released version from source, as from CRAN, with
#   no other prior setup
#   * On macOS, autobrew is used to retrieve libarrow and dependencies <---- this
```

Should this get changed to describe the new behavior this PR adds?

Reply:
ah yes good point

Reply:
I don't think we have much more docs about this than that right? Should check that ^^ maybe a follow up :see_no_evil:

Reply:
I opened #37945 for this.